File: backend/app/schemas/marketplace.py
from datetime import datetime

from app.schemas.base import CamelModel


class MarketplaceItemOut(CamelModel):
    id: str
    category: str
    name: str
    description: str
    price_cents: int
    is_premium: bool


class InstalledMarketplaceItemOut(CamelModel):
    id: str
    item: MarketplaceItemOut
    installed_at: datetime


class InstallMarketplaceItemRequest(CamelModel):
    item_id: str

File: src/types/marketplace.ts
export type MarketplaceCategory = 'question_bank' | 'revision_plan' | 'reciter' | 'theme' | 'plugin';

export interface MarketplaceItem {
  id: string;
  category: MarketplaceCategory;
  name: string;
  description: string;
  priceCents: number;
  isPremium: boolean;
}

export interface InstalledMarketplaceItem {
  id: string;
  item: MarketplaceItem;
  installedAt: string;
}
